types/shortid/index.d.ts

```diff
@@ -1,12 +1,84 @@
-declare const shortid: ShortId;
-
-export = shortid;
-
 interface ShortId {
+    /**
+     * Returns non-sequential unique id.
+     *
+     * @example
+     * users.insert({
+     *   _id: shortid(),
+     *   name: '...',
+     *   email: '...'
+     * });
+     */
     (): string;
+
+    /**
+     * Returns non-sequential unique id.
+     *
+     * @example
+     * users.insert({
+     *   _id: shortid.generate(),
+     *   name: '...',
+     *   email: '...'
+     * });
+     */
     generate: () => string;
-    characters: (string: string) => string;
-    isValid: (id: any) => boolean;
+
+    /**
+     * Change the characters used.
+     * You must provide a string of all 64 unique characters. Order is not important.
+     * The default characters provided were selected because they are url safe.
+     *
+     * @param {string} [newCharacters="0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_-"]
+     *
+     * @example
+     * // use $ and @ instead of - and _
+     * shortid.characters('0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ$@');
+     * // any 64 unicode characters work, but I wouldn't recommend this.
+     * shortid.characters('ⒶⒷⒸⒹⒺⒻⒼⒽⒾⒿⓀⓁⓂⓃⓄⓅⓆⓇⓈⓉⓊⓋⓌⓍⓎⓏⓐⓑⓒⓓⓔⓕⓖⓗⓘⓙⓚⓛⓜⓝⓞⓟⓠⓡⓢⓣⓤⓥⓦⓧⓨⓩ①②③④⑤⑥⑦⑧⑨⑩⑪⑫');
+     */
+    characters: (newCharacters: string) => string;
+
+    /**
+     * Check to see if an id is a valid shortid.
+     * Note: This only means the id could have been generated by shortid, it doesn't guarantee it.
+     *
+     * @example
+     * shortid.isValid('41XTDbE'); // true
+     * shortid.isValid('i have spaces'); // false
+     */
+    isValid: (id: unknown) => id is string;
```

Comment:
This is not a good idea; this will cause this code to behave strangely:

```ts
declare let str: string;

if (shortid.isValid(str)) {
    str // string
} else {
   str // never???
}
```

Just use a regular function.

Reply:
Done.